doc/tools/create_changelog.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+import sys
+import io
+from subprocess import check_output, CalledProcessError
+import json
+try:
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urlencode
+try:
+    from urllib2 import Request, urlopen
+except ImportError:
+    from urllib.request import Request, urlopen
+
+# this date needs to be updated with the date of the previous release
+LastReleaseDate = '2014-9-30'
```

Comment:
Hmm.... I'm not sure about this date thing.  Might a commit be better?  Don't we want to see all of the commits between LastReleaseCommit and HEAD on the current branch which we are building.